lib/async_hooks.js

```diff
@@ -224,7 +224,11 @@ class AsyncResource {
     if (typeof fn !== 'function')
       throw new ERR_INVALID_ARG_TYPE('fn', 'Function', fn);
     const ret =
-      FunctionPrototypeBind(this.runInAsyncScope, this, fn, thisArg || this);
+      FunctionPrototypeBind(
+        this.runInAsyncScope,
+        this,
+        fn,
+        arguments.length === 2 ? thisArg : this);
```

Comment:
Another option is to use a default parameter `thisArg = this`. The question is whether we want `undefined` to default to `this` or not.

Reply:
I personally think it's better to use a default parameter. Checking `arguments.length` can be problematic if someone writes a wrapper that always passes two arguments